src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12442,8 +12442,11 @@ void func_8084ECA4(Player* this, PlayState* play) {
         }
     }
 
+    //! @bug because func_8083C0E8 will change the animation and execution passes through here afterwards, this check
+    //! can pass for a single frame of the new animation the final time this function runs in a swing. This means that
+    //! the bottle can be detected as swinging for one frame well the active-looking part of the swinging animation.
```

Comment:
the last sentence is missing a word after "one frame well"

If I can suggest entirely different wording,
```c
    //! @bug The animation is changed above by func_8083C0E8, which changes curFrame to 0.
    //! This causes the flag to be set for one frame at a time when it does not look like Link is swinging the bottle.
```

Reply:
agreed on the new wording

also, I believe `func_808322D0` in the "caught something in the bottle" case will cause the same glitch. less impactful though because you will be in a cutscene during it. it may be better to just not reference the specific function and say "when your animation is changed at any point above", or something to that effect